src/Nethermind/Nethermind.Synchronization.Test/FastSync/ReceiptsSyncFeedTests.cs

```diff
@@ -231,6 +231,20 @@ public async Task When_configured_to_skip_receipts_then_finishes_immediately()
             _measuredProgressQueue.HasEnded.Should().BeTrue();
         }
 
+        [Test]
+        public async Task When_finished_sync_with_old_default_barrier_then_finishes_imedietely()
```

Comment:
tests for ReceiptsSyncFeed, but no tests for bodies
no tests for every possible case

Reply:
There are no tests for bodies in the first place. which was surprising tbh

Reply:
Added more cases for receipts